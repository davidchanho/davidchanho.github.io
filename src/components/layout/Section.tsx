import SectionHeader from 'components/layout/SectionHeader'
import React from 'react'
import { Col, Container, Row } from 'react-bootstrap'
import { SectionProps } from 'types/types'

function Section({ header, children }: SectionProps) {
    return (
        <Container className='my-5 d-flex flex-column justify-content-center'>
            <Row>
                <Col sm={3} className='border-right'>
                    <SectionHeader header={header} />
                </Col>
                <Col>
                    {children}
                </Col>
            </Row>
        </Container>
    )
}

export default Section
