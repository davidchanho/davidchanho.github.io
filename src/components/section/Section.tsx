import Header from 'components/header/Header';
import { SectionContainer } from 'components/section/Section.styles';
import { SectionProps } from 'components/section/Section.types';
import React from 'react';
import { Container } from 'react-bootstrap';

function Section({ name, header, children }: SectionProps) {
    return (
        <SectionContainer name={name}>
            <Header header={header} />
            <Container>
                {children}
            </Container>
        </SectionContainer>
    )
}

export default Section
