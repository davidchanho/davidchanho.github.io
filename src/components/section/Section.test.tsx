import React from "react";
import renderer from "react-test-renderer";
import Section from ".";

const test = {
  name: "name",
  header: "header",
  children: <div>test</div>,
};

describe("Section", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Section {...test} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
