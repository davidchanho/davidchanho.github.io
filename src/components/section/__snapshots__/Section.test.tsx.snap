// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section renders correctly 1`] = `
<div
  className="sc-bdfBwQ fKNFrQ"
  id="name"
>
  <h3
    className="text-center text-capitalize mb-5"
  >
    <u>
      header
    </u>
  </h3>
  <div
    className="container"
  >
    <div>
      test
    </div>
  </div>
</div>
`;
